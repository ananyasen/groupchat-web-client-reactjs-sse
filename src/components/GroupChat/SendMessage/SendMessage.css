.send-container{
    display:flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    padding:10px 10px 10px 20px;
}

.send-message{
    width: 100%;
    display:flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
}

.message-box{
    width: 85%;
    height: 30px;
    margin:10px 30px 10px 0px;
    border-radius: 5px;
    border: #e3e3e3 solid 1px;
    padding:8px 15px;
    font-size: 11pt;
}

.send{
    font-weight: bold;
    font-size: 13pt;
    cursor: pointer;
    color: #3b85c0;
}

.field-validation-error{
    color:darkred;
    font-size: 8pt;
}