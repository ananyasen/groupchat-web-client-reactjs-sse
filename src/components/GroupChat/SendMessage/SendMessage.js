import React, { useState } from 'react'
import './SendMessage.css'

function SendMessage({sendMessage}){

    const [message, setMessage] = useState(null)
    const [errors, setErrors] = useState({
        message : null
    })

    const onSend = (event) => {
        event.preventDefault()

        //field validation
        if(!message){
            setErrors({
                message : 'Please type a message!'
            })
        }else{
            sendMessage(message)
            setMessage('')
        }
    }

    const onEnter = (event) => {
        //clear error
        setErrors({
            message : ''
        })

        if(event.key === 'Enter'){
            onSend(event)
        }
    }

    return (
        <div className="send-container">
            <div className="send-message">
                <input type="text" value={message} placeholder="Type a message..."
                       onChange={e => setMessage(e.target.value)}
                       onKeyPress={onEnter}
                       noValidate
                       className="message-box"
                       data-testid="send-text"
                />
                <span onClick={onSend} className="send" data-testid="send-btn">Send</span>
            </div>

            {errors.message && errors.message.length > 0 &&
            <span className='field-validation-error'>{errors.message}</span>}
        </div>
    )
}

export default SendMessage
