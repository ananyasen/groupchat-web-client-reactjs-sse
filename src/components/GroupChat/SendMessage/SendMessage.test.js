import React from 'react';
import { render } from '@testing-library/react';
import SendMessage from './SendMessage'

test('SendMessage', async (done) => {
    const {container, getByText, getByTestId} = render(<SendMessage/>);
    const buttonElement = getByText(/Send/i);
    expect(buttonElement).toBeInTheDocument();

    expect(container).toMatchSnapshot()

    done()
})