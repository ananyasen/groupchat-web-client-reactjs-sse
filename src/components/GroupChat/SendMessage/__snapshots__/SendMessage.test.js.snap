// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendMessage 1`] = `
<div>
  <div
    class="send-container"
  >
    <div
      class="send-message"
    >
      <input
        class="message-box"
        data-testid="send-text"
        novalidate=""
        placeholder="Type a message..."
        type="text"
        value=""
      />
      <span
        class="send"
        data-testid="send-btn"
      >
        Send
      </span>
    </div>
  </div>
</div>
`;
